### ISSUE TITLE
[FEATURE] - Animated value interpolation.

### ISSUE THREAD

**Issue by ShanonJackson** [framer/motion#547]
**Is your feature request related to a problem? Please describe.**
The problem is useMotionValue, useTransform and declarative inline animations cannot be composed togeather;
I.E
I have opacity animating 0 -> 1
I have x animating 25% -> 50%;
I want to create a transform animation interpolating those two values as they change.
```
transform: translateX(x) rotate({opacity}deg}}
```

Interpolating values is also useful for setting a value to a "final" value when an animation reaches its end-state.

I.E
```
<motion.div initial={{height: "50px"}} animate={{height: "100px"}} interpolate={(h) => h === "100px" ? "auto" : h}>
```

Finally interpolating values is also useful in combining  multiple animating values into SVG paths; Where i challenge anyone to implement something like this  in FramerMotion where you have two independant animating values combined.
```
import { animated, interpolate } from "react-spring";
import { useSpring } from "react-spring";
function MotionValueTest({ value }: { value: number }) {
        /* 1 Atomic animating value infinitly going between 0 / 300 */
	const { x } = useSpring({
		from: {
			x: 0,
		},
		to: async (next: any) => {
			while (true) {
				await next({ x: 300 });
				await next({ x: 0 });
			}
		},
	});
        /* 1 Atomic animating value infinitly going between 50 / 100 */
	const { l } = useSpring({
		from: {
			l: 50,
		},
		to: async (next: any) => {
			while (true) {
				await next({ l: 100 });
				await next({ l: 50 });
			}
		},
	});
	return (
		<div style={{ width: "50rem", height: "20rem", border: "1px solid black" }}>
			<svg viewBox={"0 0 100 300"} height={"100%"} width={"100%"}>
				<animated.circle cx={x} cy={50} r={5} fill={"red"} />
				<animated.path
					d={interpolate([x, l], (x, l) => {
                                               /* COMBINE two atomic animating values */
						return `M ${x} 50 L ${l} ${l}`;
					})}
					stroke={"red"}
					strokeWidth={3}
				/>
			</svg>
		</div>
	);
}
```
Which produces this: 
https://imgur.com/a/SyNjrY4

**Describe the solution you'd like**
Some kind of API for interpolating multiple "animating" values togeather; This would include mixing in things like useMotionValue and useTransform which produce ATOMIC values which cannot be used to create new PRODUCT values.

**Describe alternatives you've considered**
I'm new to FramerMotion so there may be something that already exists however i cannot find it in the documentation, i'm currently on a journey to refactor out all my react-spring stuff to Framer Motion however i'm stuck on this part.

TL;DR Combine animating values the same way they offer in react-spring, love the library i'm not promoting react-spring i'm trying to move here from it and i can't refactor out this last part.

**Comment by InventingWithMonster** [comment]
Maybe it's possible to leverage template literals to make something like:

```javascript
const x = useMotionValue(0)
const opacity = useMotionValue(1)
const transform = useTransform`translateX(${x}) rotate(${opacity}deg}`
```

**Comment by InventingWithMonster** [comment]
Here's a function you can copy/paste into your codebase to try out this idea
https://gist.github.com/InventingWithMonster/38d7cdd3557f398c2ca4a2c81defed08

**Comment by ShanonJackson** [comment]
What would it look like if i needed to merge those motion values into the `initial` prop?

```ts
const ONE = useMotionValue(0);
const TWO = useMotionValue(0);
<motion.path
          initial={{d: `M ${ONE} ${TWO} L ${ONE} ${TWO}`}}
          animate={{d: M ${ONE} ${TWO} L 100 150}}
/>
// somewhere down here i have another "thing" using ONE,TWO animating back/forth that i want to join this path to and sync it with   (using onUpdate)
```

Its really hard to give small examples because this is obviously a simplified model of the a real tangible problem in my real graph library; Maybe i can publish an alpha just quickly so you can see the real problem not some trimmed down example; Because i feel like when anyone sees this example they're first thought would be "just use pathLength" and set the start `L` to the end `L` and it will achieve the same effect. However in my real use case i cannot do this its simply the smallest possible thing i can whip up to demonstrate something similar to the effect i'm trying to get.

**Comment by InventingWithMonster** [comment]
Any implementation would literally be a variation of 

```
initial={{d: `M ${ONE.get()} ${TWO.get()} L ${ONE.get()} ${TWO.get()}`}}
```

anyway, so you could do that?

**Comment by ShanonJackson** [comment]
Couldn't in my exact situation but found a much much cleaner work around without useMotionValues. Been using react-spring now for almost 2 years so its hard to get out of the normal interpolation way of thinking. Last 2 components left before the refactor is over.

Here's the end result real-time line graph i colored each of the individual lines a different color just to show the effect i was trying to obtain.

I wanted the very last line to animate from the tip of the MOVING previous last line and grow.
https://imgur.com/a/Ggvg7F1

Thanks for just getting back to me both on discord and on github really appreciate it.

**Comment by linden-dg** [comment]
@ShanonJackson apologies, I know this is over a year old, but do you remember how you solved this without useMotionValues?
